### ISSUE TITLE
array of ranges?

### ISSUE THREAD

**Issue by cwseys** [deanwilson/puppet-ip_in_range#7]
Would be great if one could pass an array of ip ranges.  :)

**Comment by cwseys** [comment]
I created a puppet Lambda function which uses ip_in_range to decide if an ip is in any of multiple subnets.
```
$ip_in_subnets=$allsubnets.reduce(false) | $truthv, String $subnet | {
   ip_in_range($::ipaddress, $subnet) or $truthv
}
```
(Sadly, only using puppet 3.8 so it appears I cannot make a puppet function of this...)
If I knew more Ruby+Puppet, what I would do is rename ip_in_range to ip_in_range_orig, create a helper function (ip_in_range) that :
- convert arg[1] in to an array, if not already
- call ip_in_range_orig for every element of the array, and return "True" if ip_in_range_orig ever returns true
- (let ip_in_range_orig do the verification of each array elements as it is called)

Thanks!

**Comment by deanwilson** [comment]
Hi @lickdragon 
Thanks for the feedback, can you try the revised function in https://github.com/deanwilson/puppet-ip_in_range/pull/8 and see if it addresses your use case please?

**Comment by deanwilson** [comment]
This is now fixed in #8 and pushed to the forge so I'll close this issue. Thank you for the suggestion, and testing it.

**Comment by cwseys** [comment]
Thanks!  Really useful.
C.
